core/src/main/java/org/elasticsearch/index/query/QueryRewriteContext.java

```diff
@@ -31,6 +32,7 @@
     protected final IndexSettings indexSettings;
     protected final IndicesQueriesRegistry indicesQueriesRegistry;
     protected final QueryParseContext parseContext;
+    protected FieldStatsProvider fieldStatsProvider;
 
     public QueryRewriteContext(IndexSettings indexSettings, ScriptService scriptService, IndicesQueriesRegistry indicesQueriesRegistry) {
```

Comment:
I can do this but it would mean passing the FieldStatsProvider into the `indexService.newQueryShardContext();` method and also intot he DefaultSearchContext constructor. Also the FieldStatsProvider will passed in as null whenever we create it outside of the SearchContext. Is that ok with you"?